{"tests": [{"output": "Your code produced an incorrect output. Here is the test code that was used and\nthe line containing the check that failed:\n\n    @weight(2)\n        @visibility('visible')\n        def test_foo():\n    >       assert submission.incorrect_answer() == 3\n    E       assert 2 == 3\n    E        +  where 2 = <function incorrect_answer at 0x10395d940>()\n    E        +    where <function incorrect_answer at 0x10395d940> = submission.incorrect_answer\n\n    test_submission.py:21: AssertionError\nThank you.", "visibility": "visible", "weight": 2, "score": 0}, {"output": "It looks like your code is trying to import 'some_module_that_doesnt_exist', but\nthis package does not exist on Gradescope. Below is the full traceback of the\nerror, showing where this import happened.\n\n    @weight(1)\n        def test_your_code_imported_an_unavailable_module():\n    >       submission.imports_missing_module()\n\n    test_submission.py:25: \n    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\n        def imports_missing_module():\n    >       import some_module_that_doesnt_exist\n    E       ModuleNotFoundError: No module named 'some_module_that_doesnt_exist'\n\n    submission.py:11: ModuleNotFoundError\nThank you.", "visibility": "after_due_date", "weight": 1, "score": 0}, {"output": "Your code unexpectedly raised ZeroDivisionError('division by zero'). Here is the\nfull traceback showing where this exception occurred.\n\n    def test_your_code_raised_an_exception():\n    >       submission.raises_exception()\n\n    test_submission.py:28: \n    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\n        def raises_exception():\n    >       1/0\n    E       ZeroDivisionError: division by zero\n\n    submission.py:8: ZeroDivisionError\nThank you.", "visibility": "after_due_date", "weight": 2, "score": 0}, {"output": "\nThank you.", "visibility": "after_due_date", "weight": 3, "score": 3}, {"output": "Your code unexpectedly raised NameError(\"name 'textwrap' is not defined\"). Here\nis the full traceback showing where this exception occurred.\n\n    def test_doctests():\n    >       run_doctests(submission)\n\n    test_submission.py:35: \n    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\n    module = <module 'submission' from '/Users/eldridge/workbench/autogradescope/example/submission.py'>\n\n        def run_doctests(module):\n            result = doctest.testmod(module)\n            if result.failed > 0:\n    >           msg = textwrap.dedent(f\"\"\"\n                    Your code ran, but some of the doctests failed. Make sure to\n                    check the doctests by running them on your own machine.\n                    \"\"\".strip('\\n'))\n    E           NameError: name 'textwrap' is not defined\n\n    autogradescope/__init__.py:9: NameError\nThank you.", "visibility": "after_due_date", "weight": 2, "score": 0}, {"output": "Your code unexpectedly raised TimeoutError('Your code took longer than 3 seconds\nto run, which is too long. We have kindly asked your code to stop running.\\n').\nHere is the full traceback showing where this exception occurred.\n\n    @timeout(3)\n        def test_timeout():\n    >       submission.i_take_4ever()\n\n    test_submission.py:39: \n    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n    submission.py:30: in i_take_4ever\n        pass\n    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\n    args = (14, <frame at 0x10394d4c0, file '/Users/eldridge/workbench/autogradescope/example/submission.py', line 30, code i_take_4ever>)\n    kwargs = {}\n\n        def handler(*args, **kwargs):\n    >       raise TimeoutError(msg)\n    E       autogradescope.exceptions.TimeoutError: Your code took longer than 3 seconds to run, which is too long. We have kindly asked your code to stop running.\n\n    autogradescope/_timeout.py:26: TimeoutError\nThank you.", "visibility": "after_due_date", "weight": 2, "score": 0}]}